### PR TITLE
Support git origin URLs that start with org-

### PR DESCRIPTION
**Pull request by timtebeek**
Fixes https://github.com/moderneinc/moderne-cli/issues/319

### CODE REVIEW

**Review by rpau — CHANGES_REQUESTED** [review]
`org-` is not a URL pattern, so this solution is not robust. The requirement to solve the issue is to support user@host.xz:/path/to/repo.git/ urls

File: rewrite-core/src/main/java/org/openrewrite/marker/GitProvenance.java [hunk 1]
```diff
@@ -95,21 +97,20 @@ public String getOrganizationName() {
...
+                path = path.substring(1);
+            }
+            int firstIndexOfSlash = path.indexOf('/');
+            int lastIndexOfSlash = path.lastIndexOf('/');
+            if (firstIndexOfSlash != lastIndexOfSlash) {
+                return path.substring(firstIndexOfSlash + 1, lastIndexOfSlash);
+            }
+            return path.substring(0, lastIndexOfSlash);
+        } catch (URISyntaxException e) {
+            return null;
```
Comment (timtebeek) [inline rewrite-core/src/main/java/org/openrewrite/marker/GitProvenance.java:113]: Wasn't sure about returning null here & in getRepositoryName

File: rewrite-core/src/test/java/org/openrewrite/marker/GitProvenanceTest.java [hunk 2]
```diff
@@ -57,7 +57,9 @@ private static Stream<String> remotes() {
           "https://github.com/openrewrite/rewrite.git",
           "file:///openrewrite/rewrite.git",
           "http://localhost:7990/scm/openrewrite/rewrite.git",
-          "git@github.com:openrewrite/rewrite.git"
+          //TODO "http://localhost:7990/scm/some/openrewrite/rewrite.git",
```
Comment (timtebeek) [inline rewrite-core/src/test/java/org/openrewrite/marker/GitProvenanceTest.java:60]: This is a case where we actually need a baseUrl to determine which is which.

File: rewrite-core/src/main/java/org/openrewrite/marker/GitProvenance.java [hunk 3]
```diff
@@ -95,21 +97,20 @@ public String getOrganizationName() {
...
-            return "";
+        try {
+            URIish gituri = new URIish(origin);
+            String path = gituri.getPath();
+            if (path.startsWith("/")) {
+                path = path.substring(1);
+            }
+            int firstIndexOfSlash = path.indexOf('/');
+            int lastIndexOfSlash = path.lastIndexOf('/');
+            if (firstIndexOfSlash != lastIndexOfSlash) {
```
Comment (timtebeek) [inline rewrite-core/src/main/java/org/openrewrite/marker/GitProvenance.java:108]: This isn't the best, as it fails on additional slashes such as:
`http://localhost:7990/scm/some/openrewrite/rewrite.git`
Those cases now also already requite a baseUrl, and are common in GitLab.

File: rewrite-core/src/main/java/org/openrewrite/marker/GitProvenance.java [hunk 4]
```diff
@@ -87,29 +89,21 @@ public String getOrganizationName(String baseUrl) {
...
-        int secondSlashPos = path.lastIndexOf('/', firstSlashPos - 1);
-
-        if (secondSlashPos > -1) {
-            return path.substring(secondSlashPos + 1, firstSlashPos);
-        } else if (firstSlashPos > -1) {
-            return path.substring(0, firstSlashPos);
-        } else {
-            return "";
+        try {
+            String path = new URIish(origin).getPath();
```
Comment (rpau) [inline rewrite-core/src/main/java/org/openrewrite/marker/GitProvenance.java:100]: nice!

**Review by rpau — APPROVED** [review]
LGTM

**Pull request merged**